src/Datasource/QueryInterface.php

```diff
@@ -258,14 +257,22 @@ public function page(int $num, ?int $limit = null);
     public function toArray(): array;
 
     /**
-     * Returns the default repository object that will be used by this query,
-     * that is, the repository that will appear in the from clause.
+     * Set the default Table object that will be used by this query
+     * and form the `FROM` clause.
      *
      * @param \Cake\Datasource\RepositoryInterface $repository The default repository object to use
      * @return $this
      */
     public function repository(RepositoryInterface $repository);
```

Comment:
Shall we call it `setRepository()`?

Reply:
in other cases we dont have setX methods for query methods , and just `x()` for setter  and `getX()`

Reply:
yep, I remeber this now 👍